libs/velo-external-db-core/src/web/jwt-auth-middleware.ts

```diff
@@ -0,0 +1,71 @@
+import { errors } from '@wix-velo/velo-external-db-commons'
+const { UnauthorizedError } = errors
+import { JwtPayload, Secret, verify } from 'jsonwebtoken'
+import * as express from 'express'
+import { WixDataFacade } from './wix_data_facade'
+
+
+export const TOKEN_ISSUER = 'wix-data.wix.com'
+
+export class JwtAuthenticator {
+    publicKey: string | undefined = undefined
+    externalDatabaseId: string
+    allowedMetasites: string[]
+    wixDataFacade: WixDataFacade
+
+    constructor(externalDatabaseId: string, allowedMetasites: string, wixDataFacade: WixDataFacade) {
+        this.externalDatabaseId = externalDatabaseId
+        this.allowedMetasites = allowedMetasites.split(',')
+        this.wixDataFacade = wixDataFacade
+    }
+
+    authorizeJwt() {
+        return async(req: express.Request, res: express.Response, next: express.NextFunction) => {
+            try {
+                const token = this.extractToken(req.header('Authorization'))
+                if (this.publicKey === undefined) {
+                    this.publicKey = await this.wixDataFacade.getPublicKey(this.externalDatabaseId)
+                }
+                await this.verify(token)
+            } catch (err) {
+                // @ts-ignore
+                console.error('Authorization failed: ' + err.message)
+                next(new UnauthorizedError('You are not authorized'))
+            }
+            next()
+        }
+    }
+
+    async verifyWithRetry(token: string): Promise<JwtPayload | string> {
+        try {
+            return verify(token, this.publicKey as Secret)
+        } catch (err) {
+            this.publicKey = await this.wixDataFacade.getPublicKey(this.externalDatabaseId)
+            return verify(token, this.publicKey as Secret)
+        }
+    }
+
+    async verify(token: string) {
+        const { iss, metasite } = await this.verifyWithRetry(token) as { iss: string, metasite: string }
+
+        if (iss === undefined || iss !== TOKEN_ISSUER) {
```

Comment:
I think it should be `as JwtPayload `. 
I don't think it's right to declare it as a string and then check if it's undefined. JwtPayload mark these as optional, and then it makes more sense.

Reply:
if TOKEN_ISSUER is a constant so checking if `iss` is undefined is redundant